### PR TITLE
Feature/jkw to read list

### PR DESCRIPTION
**Pull request by kip-west**
#### What's this PR do?
- Adds a `toReadList` to books state & buttons to each card to add the book object to this list. The buttons have preliminary styling (turning the text red & green based on `active`/`inactive` in the classList).

Type of change made:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Styling

#### Where should the reviewer start?
- Check out these files in tandem.

#### Why is this change required? What problem does it solve?
- We needed a way for users to add books to their reading list. This adds state & functionality to add/remove these books.

#### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
- It'd be great to have some eyes on the logic I used to determine if a book is already on the list! This function feels bulky & I'm not sure if we should break it into smaller pieces.

#### What are the relevant tickets?
- #10 (still need to render different Book Containers to finish this ticket)

#### Screenshots (if appropriate)
- jkw-readingList-preliminary

#### Questions:
- As we talked about this morning, we need to figure out how to reuse our Book Container component to make our separate lists! We also need to create out library in state that holds all books...

### CODE REVIEW

File: src/Components/Book/Book.js [hunk 1]
```diff
@@ -3,13 +3,16 @@ import ReactDOM from 'react-dom';
...
-const Book = ({book}) => {
+const Book = ({ book, addBook }) => {
   return (
     <section class="book-card-section">
         <h1>{book.title}</h1>
         <h3>{book.author}</h3>
         <h3>Ranking: {book.rank}</h3>
         <img className="Book-card-image" alt="Book cover" src={book.book_image} />
+        <button id={book.primary_isbn10} onClick={addBook} className="reading-list-button active">
+          Reading List
```
Comment (NicoleLatifi) [inline src/Components/Book/Book.js:17]: Make we make the text for this button an action like "Add to Reading List" or "Read"? I used "Add to Reading List" on the book info page, so maybe "Read" would work. Do you think a user would understand what the button does if it just said "Read"?

Comment (kip-west) [reply src/Components/Book/Book.js:17]: Completely agreed!!

**Review by NakiNorton — APPROVED** [review]

File: src/Components/Books/Books.js [hunk 2]
```diff
@@ -20,9 +21,36 @@ class Books extends Component {
       .catch(error => alert(error.message))
   }
 
+  changeButtonStyling(id) {
```
Comment (NicoleLatifi) [inline src/Components/Books/Books.js:26]: This works. We could consider making the book component a class component with state that holds whether or not the book is added to the list. Then we wouldn't have to use classList.contains to determine if a book is added to the list. But I'm not sure which way would be better.

**Review by NicoleLatifi — APPROVED** [review]

File: src/Components/Books/Books.js (hunk 2 above)
Comment (amykarnaze) [reply src/Components/Books/Books.js:26]: Great work, Jake! My only thoughts are using a toggle to reduce a line of code.

**Comment by NakiNorton** [comment]
This looks great! Once we have implemented Redux we'll have time to look at opportunities for refactoring. Thanks for merging these branches too, never an easy task.

**Review by amykarnaze — APPROVED** [review]

**Comment by amykarnaze** [comment]
Great work, Jake! My only thoughts are using a toggle to reduce a line of code.

**Pull request merged**